operators/Makefile

```diff
@@ -280,39 +280,38 @@ TESTS_MATCH ?= ""
 E2E_IMG ?= $(IMG)-e2e-tests:$(TAG)
 
 # Run e2e tests as a k8s batch job
-e2e: e2e-setup e2e-deploy-operators e2e-docker-image e2e-run
+e2e: docker-build docker-push e2e-docker-build e2e-docker-push e2e-run
 
-e2e-setup:
-	@ echo -- Setup e2e rbac
-	kubectl apply -f config/e2e/rbac.yaml
-
-e2e-deploy-operators:
-	@ echo -- Deploy operators to manage e2e namespace
-	@ $(MAKE) --no-print-directory MANAGED_NAMESPACE=e2e deploy
-
-e2e-docker-image:
-	@ echo -- Build and push e2e docker image
+e2e-docker-build:
 	docker build -t $(E2E_IMG) -f test/e2e/Dockerfile .
+
+e2e-docker-push:
 	docker push $(E2E_IMG)
 
 e2e-run:
-	@ echo -- Run e2e as a job
-	./hack/run-e2e.sh "$(E2E_IMG)" "$(TESTS_MATCH)"
+	@go run test/e2e/cmd/main.go run \
+		--operator-image=$(OPERATOR_IMAGE) \
+		--e2e-image=$(E2E_IMG) \
+		--test-regex=$(TESTS_MATCH)
 
 # Verify e2e tests compile with no errors, don't run them
 e2e-compile:
 	go test ./test/e2e/... -run=dryrun > /dev/null
 
 # Run e2e tests locally (not as a k8s job), with a custom http dialer
 # that can reach ES services running in the k8s cluster through port-forwarding.
-e2e-local: e2e-setup e2e-run-local
+e2e-local: docker-build docker-push e2e-run-local
```

Comment:
I don't think we need to docker-build and push here, do we?

Reply:
The tests still run on a Kubernetes cluster so the operator images should be pushed to the associated registry. Previously it was done by the `e2e-setup` target that I have now broken down into smaller targets.

I can see why it's confusing because "local" implies that everything is local. I didn't try to change the name of the target because I thought people might be used to that name by now.

Reply:
My main workflow is to run both the operator and the e2e tests locally when I work on something that may break one of the e2e tests. So I don't need to build nor push a Docker image. I don't know about others?

Reply:
Ah, I seem to have misread the old Makefile because I was convinced it was pushing the operator image even for local tests. I will change the local run target to match the workflow you describe.